crates/ruff/src/checkers/noqa.rs

```diff
@@ -161,11 +161,16 @@ pub fn check_noqa(
                                     Location::new(row + 1, start - leading_spaces),
                                     Location::new(row + 1, end + trailing_spaces),
                                 ));
-                            } else {
+                            } else if lines[row].chars().nth(end + 1).map_or(false, |c| c == '#') {
                                 diagnostic.amend(Fix::deletion(
                                     Location::new(row + 1, start),
                                     Location::new(row + 1, end + trailing_spaces),
                                 ));
+                            } else {
+                                diagnostic.amend(Fix::deletion(
+                                    Location::new(row + 1, start + 1),
+                                    Location::new(row + 1, end),
+                                ));
```

Comment:
Is my understanding correct that we'll change `#noqa comment` to `# comment`. What's the reasoning behind this choice?

Reply:
Yeah this seems wrong -- will change.